api/providers/tasks.py

```diff
@@ -453,10 +453,16 @@ def handle_registration_row(row, initiator, provider, schema, auto_approval=Fals
         )
         # Remove all contributors except the initiator if created from an existing node
         if node:
+            # Temporarily make initiator contributor visible so that removal of the others can succeed.
+            initiator_contributor = draft.contributor_set.get(user=initiator)
+            if not initiator_contributor.visible:
+                initiator_contributor.visible = True
+                initiator_contributor.save()
             contributor_set = draft.contributor_set.all()
             for contributor in contributor_set:
                 if initiator != contributor.user:
-                    draft.remove_contributor(contributor, auth)
+                    is_removed = draft.remove_contributor(contributor, auth)
+                    assert is_removed, 'Removal of an non-initiator contributor from the draft has failed'
```

Comment:
Since task don't control the removal process, thus added an assertion for redundancy in case contributors are not removed due to other unexpected causes.